### PR TITLE
Add forum routes authentication

### PR DESCRIPTION
**Pull request by namandangi**
tested entire crud with postman

### CODE REVIEW

**Comment by namandangi** [comment]
I actually have used req.user in forumAuth.js . In forum controller I first call the auth middleware which initializes the req.user and then I call forumAuth middleware(i.e owner) which checks if req.user._id is equal to the id of the user that posted the particular forum

**Comment by vixrant** [comment]
But when you are posting a new forum post, you should set the poster = req.user, don't take poster from the request data. That way user A can insert a post as user B.

**Comment by vixrant** [comment]
Instead of that, mention it explicitly that poster = req.user. Don't ask the frontend to send the client user id, as they are already sending a token for authorisation.

File: infra/middleware/forumAuth.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+            return next();
+    }
+    catch(err)
+    {
+        console.log(err);
+        res.status(401).send({error:'Invalid Request!'})
+    }
+}
+
+module.exports = {owner};
```
Comment (vixrant) [inline infra/middleware/forumAuth.js]: Name the function better.

`isOwnerOfForum`.

**Pull request merged**